digitalpy/core/zmanager/impl/zmq_pusher.py

```diff
@@ -0,0 +1,44 @@
+import zmq
+from digitalpy.core.zmanager.pusher import Pusher
+from digitalpy.core.zmanager.request import Request
+from digitalpy.core.parsing.formatter import Formatter
+
+class ZMQPusher(Pusher):
+    
+    def __init__(self, formatter: Formatter):
+        self.pusher_context = None
+        self.pusher_socket = None
+        self.pusher_formatter = formatter
+
+    def initiate_connections(self, port: int, address: str):
+        # added to fix hanging connect issue as per 
+        # https://stackoverflow.com/questions/44257579/zeromq-hangs-in-a-python-multiprocessing-class-object-solution
+        if self.pusher_context == None:
+            self.pusher_context = zmq.Context()
+        if self.pusher_socket == None:
+            self.pusher_socket = self.pusher_context.socket(zmq.PUSH)
+        self.pusher_socket.connect(f"tcp://{address}:{port}")
+
+    def subject_bind(self, address: int, port: str):
+        """create the ZMQ zocket
```

Comment:
**[pep8]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>D200 One-line docstring should fit on one line with quotes